source/AutoReviser.Tests/ImmutableExtensions_specs.cs

```diff
@@ -325,5 +359,45 @@ public class HasImmutableArrayProperty
 
             revision.Delta[1].Echo.Alfa.Should().Be(element);
         }
+
+        [TestMethod]
+        [AutoData]
+        public void Revise_updates_element_with_internal_constructor(
```

Comment:
IMO this method tests that updating properties, not elements. So I recommend that you change the name of the test method.